nixos/modules/services/misc/ssm-agent.nix

```diff
@@ -43,5 +41,26 @@ in {
         RestartSec = "15min";
       };
     };
+
+    # Add user that Session Manager needs, and give it sudo.
+    # This is consistent with Amazon Linux 2 images.
+    security.sudo.extraRules = [
+      {
+        users = [ "ssm-user" ];
+        commands = [
+          {
+            command = "ALL";
+            options = [ "NOPASSWD" ];
+          }
+        ];
+      }
+    ];
+    # On Amazon Linux 2 images, the ssm-user user is pretty much a
+    # normal user with its own group. We do the same.
+    users.groups.ssm-user = {};
+    users.users.ssm-user = {
+      isNormalUser = true;
+      group = "ssm-user";
```

Comment:
Is there a point to using a separate user if that user has the same privileges as root? Why not just run as root directly?

Reply:
Good question. The user is not used to run the service, it's used when you use Session Manager to connect directly to a running instance.

Reply:
(This does not appear to be a configurable thing, and is how it happens on standard AWS images too.)